syntax = "proto2";

package distributed_estimation;

message Vector3 {
    optional float x = 1;
    optional float y = 2;
    optional float z = 3;
}

message Quaternion {
    optional float x = 1;
    optional float y = 2;
    optional float z = 3;
    optional float w = 4;
}

message Transform {
    optional Vector3 translation = 1;
    optional Quaternion rotation = 2;
}

message FleetPlanning {
    optional string robot_name = 1;
    optional Transform transform = 2;
    optional uint32 status_info = 3;

}